Sources/ComposableArchitecture/Observation/NavigationStack+Observation.swift

```diff
@@ -204,6 +209,7 @@ public struct _NavigationDestinationViewModifier<
       .navigationDestination(for: StackState<State>.Component.self) { component in
         navigationDestination(component: component)
           .environment(\.navigationDestinationType, State.self)
+          .environment(\.stackElementGeneration, component.id.generation + 1)
```

Comment:
The `NavigationLink` is created within the `View.body`, and it appears that the only safe way to increase the stack offset is through the `environment`.